### ISSUE TITLE
Configure fails if CMAKE_C_STANDARD is not specified

### ISSUE THREAD

**Issue by antonio-rojas** [LLNL/sundials#96]
If `CMAKE_C_STANDARD` is not explicitly specified in the cmake command, configure fails with
```
CMake Error at /build/sundials/src/build/KLU_TEST/CMakeLists.txt:13 (ADD_EXECUTABLE):
  C_STANDARD is set to invalid value ''


CMake Error at cmake/tpl/SundialsKLU.cmake:95 (try_compile):
  Failed to generate test project build system.
Call Stack (most recent call first):
  cmake/SundialsSetupTPLs.cmake:40 (include)
  CMakeLists.txt:170 (include)
```
cmake 3.22.1, sundials 6.0.0

**Comment by balos1** [comment]
Thanks for reporting this. In the meantime a workaround is to just set `CMAKE_C_STANDARD`. as you mention. Setting the `CMAKE_C_STANDARD` is known to break the HIP enabled examples in SUNDIALS however. So, if you are using `ENABLE_HIP=ON` and getting this error about `C_STANDARD`, the workaround is to disable example with `EXAMPLES_ENABLE_CXX=OFF`.

**Comment by balos1** [comment]
Fixed in SUNDIALS v6.1.0. Thanks!
